source/common/router/router.cc

```diff
@@ -178,7 +178,8 @@ FilterUtility::finalTimeout(const RouteEntry& route, Http::RequestHeaderMap& req
     }
   } else {
     const Http::HeaderEntry* header_timeout_entry = request_headers.EnvoyUpstreamRequestTimeoutMs();
-    if (trySetGlobalTimeout(header_timeout_entry, timeout)) {
+    if (header_timeout_entry) {
+      trySetGlobalTimeout(header_timeout_entry, timeout);
```

Comment:
It looks like the return value from trySetGlobalTimeout is now being ignored. In the previous code, when it returned false, the header would not be removed. But by ignoring the return value, it is always ignored. Is this expected? This looks similar to the code on line 175.

Reply:
This is expected behavior. In previous implementation, we will return `true` even if `atoi` conversion failed. It means that it will be always true when the request header has timeout entry. In addition to this fact, remove header doesn't affect any. To conclude this, to call remove header as it has time entry or not doesn't have any problem.

Reply:
Thank you for your helpful reviews. deleted the test.